libs/SalesforceSDK/src/com/salesforce/androidsdk/accounts/UserAccountManager.java

```diff
@@ -292,6 +317,7 @@ public void switchToUser(UserAccount user, int userSwitchType, Bundle extras) {
 		if (user.equals(curUser)) {
 			return;
 		}
+		clearCachedCurrentUser(curUser);
```

Comment:
This only handles the switch to existing user use-case. You also need to call this in the switch to new user use-case. I think a better place to handle this would be in `public void storeCurrentUserInfo(String userId, String orgId)` in `UserAccountManager`. Any change to current user ends up there, including new user and current user. This call can be moved there to handle both cases.

Reply:
Hmm but the only caller of storeCurrentUserInfo(...) is this method `switchToUser(UserAccount user, int userSwitchType, Bundle extras)`, how do the store values get updated in the switch to new user use-case?